src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationConstants.cs

```diff
@@ -0,0 +1,266 @@
+            /// Current user tenant identifier.
+            /// </summary>
+            public const string ContextIdentifier = "http://schemes.superoffice.net/identity/ctx";
+
+            /// <summary>
+            /// Current SuperId user email address.
+            /// </summary>
+            public const string Email = "http://schemes.superoffice.net/identity/email";
+
+            /// <summary>
+            /// Current user first name.
+            /// </summary>
+            public const string FirstName = "http://schemes.superoffice.net/identity/firstname";
+
+            /// <summary>
+            /// SuperOffice identity provider name.
+            /// </summary>
+            public const string IdentityProvider = "http://schemes.superoffice.net/identity/identityprovider";
+
+            /// <summary>
+            /// Current user initials.
+            /// </summary>
+            public const string Initials = "http://schemes.superoffice.net/identity/initials";
+
+            /// <summary>
+            /// Determines if current user is an administrator.
+            /// </summary>
+            public const string IsAdministrator = "http://schemes.superoffice.net/identity/is_administrator";
+
+            /// <summary>
+            /// Current user last name.
+            /// </summary>
+            public const string LastName = "http://schemes.superoffice.net/identity/lastname";
+
+            /// <summary>
+            /// Endpoint of SuperOffice SOAP web services.
+            /// </summary>
+            public const string NetServerUrl = "http://schemes.superoffice.net/identity/netserver_url";
+
+            /// <summary>
+            /// Tenant database serial number.
+            /// </summary>
+            public const string Serial = "http://schemes.superoffice.net/identity/serial";
+
+            /// <summary>
+            /// Current user's tenant primary email address.
+            /// </summary>
+            public const string PrimaryEmail = "http://schemes.superoffice.net/identity/so_primary_email_address";
+
+            /// <summary>
+            /// Identifier used to exchange for a system user ticket.
+            /// </summary>
+            public const string SystemToken = "http://schemes.superoffice.net/identity/system_token";
+
+            /// <summary>
+            /// Credentials token used to access web services.
+            /// </summary>
+            public const string Ticket = "http://schemes.superoffice.net/identity/ticket";
+
+            /// <summary>
+            /// Current user's username.
+            /// </summary>
+            public const string UserPrincipalName = "http://schemes.superoffice.net/identity/upn";
+
+            /// <summary>
+            /// Endpoint of SuperOffice REST web services.
+            /// </summary>
+            public const string WebApiUrl = "http://schemes.superoffice.net/identity/webapi_url";
+        }
+
+        public static class FormatStrings
```

Comment:
Do these need to be public if they're an implementation detail?

Reply:
Not the FormatStrings. I've marked that as internal now.